src/main/java/mat/cql/CqlToMatXml.java

```diff
@@ -0,0 +1,609 @@
+        destinationModel.setCodeSystemList(parseRepeatingLine(CODESYSTEM_TOKEN, s -> {
+            //codesystem "LOINC": 'urn:oid:2.16.840.1.113883.6.1'
+            ParseResult systemName = nextQuotedString(s, 0);
+            ParseResult version = nextTickedString(s, systemName.getEndIndex());
+
+            if (areValidAscendingIndexes(systemName, version)) {
+                //For now these should be identical to the sourceCqlModel versions.
+                //Just find the corresponding one by code system name and use that.
+                String sysName = systemName.getString();
+                String ver = trimUrn(version.getString());
+                log.info("sysName=" + sysName + " version=" + ver);
+                return findExisting(sourceModel.getCodeSystemList(),
+                        cs -> StringUtils.equals(systemName.getString(), cs.getCodeSystemName()),
+                        "Could not find sourceCqlModel.codeSystemList for " + sysName);
+            } else {
+                throw new IllegalArgumentException("Invalid include encountered: " + s);
+            }
+        }));
+    }
+
+    /**
+     * Parses all of the valuesets found in the cql.
+     */
+    private void processValueSets() {
+        destinationModel.setValueSetList(parseRepeatingLine(VALUESET_TOKEN, s -> {
+            //valueset "Ethnicity": 'urn:oid:2.16.840.1.114222.4.11.837'
+            ParseResult type = nextQuotedString(s, 0);
+            ParseResult version = nextTickedString(s, type.getEndIndex());
+
+            if (areValidAscendingIndexes(type, version)) {
+                //For now these should be identical to the sourceCqlModel versions.
+                //Just find the corresponding one by code system name and use that.
+                return findExisting(sourceModel.getValueSetList(),
+                        vs -> StringUtils.equals(type.getString(), vs.getCodeListName()),
+                        "Could not find sourceCqlModel.valueSetList for " + type.getString());
+            } else {
+                throw new IllegalArgumentException("Invalid valueset encountered: '" + s);
+            }
+        }));
+    }
+
+    /**
+     * Parses all of the codes out of the cql..
+     */
+    private void processCodes() {
+        destinationModel.setCodeList(parseRepeatingLine(CODE_TOKEN, s -> {
+            //code "Birth date": '21112-8' from "LOINC" display 'Birth date'
+            ParseResult name = nextQuotedString(s, 0);
+            ParseResult code = nextTickedString(s, name.getEndIndex() + 1);
+            ParseResult codeSystem = nextQuotedString(s, code.getEndIndex() + 1);
+            ParseResult displayName = nextTickedString(s, codeSystem.getEndIndex() + 1);
+
+            if (areValidAscendingIndexes(name, code, codeSystem, displayName)) {
+                //For now these should be identical to the sourceCqlModel versions.
+                //Just find the corresponding one by code system name and use that.
```

Comment:
Similiarly here, comment states 'code system name', but code is comparing 'name'.

Reply:
fixed.